src/widgets/Tabs/VaccineSelect.js

```diff
@@ -147,19 +165,65 @@ const VaccineSelectComponent = ({
 
       <FlexRow flex={1} alignItems="flex-end" justifyContent="flex-end">
         <PageButtonWithOnePress text={buttonStrings.cancel} onPress={onCancelPrescription} />
-        <PageButtonWithOnePress
-          text={buttonStrings.confirm}
-          style={{ marginLeft: 'auto' }}
-          isDisabled={!selectedBatches && !hasRefused}
-          onPress={confirmPrescription}
+
+        {wasVaccinatedToday() ? (
+          <PageButton
+            text={buttonStrings.confirm}
+            style={{ marginLeft: 'auto' }}
+            isDisabled={!selectedBatches && !hasRefused}
+            onPress={toggleConfirmDoubleDoseModal}
+          />
+        ) : (
+          <PageButtonWithOnePress
+            text={buttonStrings.confirm}
+            style={{ marginLeft: 'auto' }}
+            isDisabled={!selectedBatches && !hasRefused}
+            onPress={confirmPrescription}
+          />
+        )}
```

Comment:
Maybe just use `useDebounce` on the `confirmPrescription` callback, and use `PageButton` for both cases? 

```
const debouncedConfirmPrescription = useDebounce(confirmPrescription);
const confirmOnPress = wasVaccinatedToday() ? debouncedConfirmPrescription : toggleConfirmDoubleDoseModal`
...
onPress={confirmOnPress}
/>
```